### ISSUE TITLE
ES6: Use * to Import Everything from a File - import name ambiguity 

### ISSUE THREAD

**Issue by mstellaluna** [freeCodeCamp/freeCodeCamp#16356]
#### Challenge Name
[ES6: Use * to Import Everything from a File](https://beta.freecodecamp.org/en/challenges/es6/use--to-import-everything-from-a-file)

#### Issue Description

~The challenge fails with 'require not defined' when importing from capitalize_string using the * wildcard.~ (edited by @systimotic: this part has been resolved)

Also the instructions is missing the object name to use myStringModule as the the 'as' name so it follows the editor. Currently the examples in the lesson state to use a name of your choice but that will fail since in the editor it is setup as myStringModule.capitalize("foo");

#### Browser Information

Browser Name, Version:
Operating System: FireFox 57.0 (64-bit) and Chrome Version 63.0.3239.84 (Official Build) (64-bit)
Mobile, Desktop, or Tablet: Laptop Windows 10 Professional 64-bit

#### Your Code

```js
"use strict";
import * as myStringModule from "capitalize_strings";
myStringModule.capitalize("foo");
myStringModule.lowercase("Foo");`


```
#### Screenshot

![image](https://user-images.githubusercontent.com/22984739/34418637-1ae9e154-ebcd-11e7-8dbd-0cf7a0fc1981.png)

**Comment by systimotic** [comment]
Thanks for all of the QA work @mstellaluna!

The `require not defined` part is a duplicate of #16235 

I think the "you may use any name" portion is technically correct. You could change `myStringModule` to `potato` only in this file and everything would still work fine, regardless of names in the imported file.
The instructions below the horizontal line should clarify that it's expecting it to be named `myStringModule`.

**Comment by theoutlander** [comment]
This issue is fixed with https://github.com/freeCodeCamp/freeCodeCamp/pull/16340#event-1405641303

**Comment by systimotic** [comment]
Thanks for that fixing that @theoutlander, nice work! I'm leaving this issue open because the other part ("you may use any name") is still the same.

**Comment by raisedadead** [comment]
Hi, sorry I am a little lost here, can anyone of you guys, elaborate a little on what is to be done?

**Comment by systimotic** [comment]
@raisedadead Of course!

Here's an excerpt from the challenge description:
> You may use any name following the import * as portion of the statement.

This is true, you can give it whatever name you want.
The issue is that the challenge test requires that name to be `myStringModule`, but it does not mention this requirement anywhere. It is only suggested by the seed, but not even hinted at in the test or description.

**Comment by theoutlander** [comment]
Let's come to a consensus on what needs to be done here and I'll try to fix it when I get a chance.

**Comment by theoutlander** [comment]
I think this could also be marked as a first timer issue if it's an easy one. I am working on another issue as it is.

**Comment by raisedadead** [comment]
@systimotic got it, thanks for the additional clarification. I am sorry, I haven't looked at all of the ES6 curriculum in depth yet. 

But, this does make sense. I think we should update the tests to allow for the import to be anything.

**Comment by systimotic** [comment]
Agreed. I'm not sure if we can make the test allow any input because there is already some other code in the editor that requires a specific name, but if we can't do that, we should at least explicitly state the name required for this challenge in the description.

**Comment by ahmadabdolsaheb** [comment]
If no one is working on this i would like to write a regex test for it and see if I can make it work.

**Comment by raisedadead** [comment]
Yes, please.

**Comment by ahmadabdolsaheb** [comment]
@raisedadead, I looked into the challenge and I think @systimotic is right. Writing a test that checks for any entered value could be done, but If we allow the campers to use X with the * statement we should also check to see if X is used correctly in the lines below it. Since I cannot store the entered value, I cannot find a good way to check for it in the rest of the document. Here are my suggestions in order of preference:

1. change the description to Inform the campers that although they can use any name they choose, for this challenge they need to use a pre-determined name.
2. delete "myStringModule.capitalize(\"foo\");" and "myStringModule.lowercase(\"Foo\");" in the challenge seed so writing a test with any name would be possible. 

Also a semi-column is missing after <code>import * as myMathModule from "math_functions"</code> in the description.

Let me know what you think so I prepare a draft accordingly.

**Comment by raisedadead** [comment]
I would prefer the second option. 

> delete "myStringModule.capitalize("foo");" and "myStringModule.lowercase("Foo");" in the challenge seed so writing a test with any name would be possible.

**Comment by ahmadabdolsaheb** [comment]
We could use the following general purpose regex 
<code>\import\s+\*\s+as\s+[a-zA-Z0-9_$]+\s+from\s+"math_functions"\s*;\g</code>
This accepts for names that are made with a-z, A-Z, 0-9, _, and $. However, it also allows reserved javascript names and names that start with a number. 

you can check it here https://regex101.com/r/RkheRs/27

let me know if it is satisfactory.

**Comment by ahmadabdolsaheb** [comment]
@raisedadead We could use the following general purpose regex 
<code>\import\s+\*\s+as\s+[a-zA-Z0-9_$]+\s+from\s+"math_functions"\s*;\g</code>
This accepts for names that are made with a-z, A-Z, 0-9, _, and $. However, it also allows reserved javascript names and names that start with a number. 

check it out here https://regex101.com/r/RkheRs/27

let me know if it is satisfactory.

**Comment by QuincyLarson** [comment]
@raisedadead what do you think of @ahmadabdolsaheb's suggestion?

**Comment by raisedadead** [comment]
Hi @ahmadabdolsaheb yup sounds good to me. Lets have a PR.

**Comment by pahosler** [comment]
there is still a problem with this exercise.
the challenge states:

> The code below requires the contents of a file, "capitalize_strings", found in the same directory as it, imported. Add the appropriate import * statement to the top of the file, using the object provided.

which is fine, but it doesn't give complete information to complete the challenge, I had to look in this issue to see what is ACTUALLY expected.

the code section has one line only:
`"use strict";`

there is absolutely nothing in the body of the explanation or the challenge description telling us what is actually required in the code!

if you do what it asks by adding:
import * as myModule from "capitalize_strings";

it get an error<?> when you run the code that is less than helpful:

`uses import * as properly`

nowhere in the description does it say to use the module or what is in the module

for example:
```js
"use strict";
import * as thing from "capitalize_strings";
thing.capitalize("foo");
thing.lowercase("Bar");
```
interestingly however, now that I've had a successful test it passes with only the import line added to the code. I tried to repeat the error in firefox, originally the test was done in chrome, but again it passes the test with only the `import * as myModule from "capitalize_strings";` line

This test would not pass previously until the code above created a successful result.
